sdk/src/main/scala/com/horizen/account/abi/ABIDecoder.scala

```diff
@@ -0,0 +1,42 @@
+package com.horizen.account.abi
+
+import org.web3j.abi.datatypes.Type
+import org.web3j.abi.{DefaultFunctionReturnDecoder, TypeReference}
+
+import java.util
+import scala.util.Try
+
+object ABIDecoder {
+  val OP_CODE_LENGTH = 4
+
+  def getArgumentsFromData(data: Array[Byte]): Array[Byte] = {
+    if (data.length < ABIDecoder.OP_CODE_LENGTH) throw new IllegalArgumentException("Data length " + data.length + " must be >= " + ABIDecoder.OP_CODE_LENGTH)
+    util.Arrays.copyOfRange(data, ABIDecoder.OP_CODE_LENGTH, data.length)
+  }
+
+  def getOpCodeFromData(data: Array[Byte]): Array[Byte] = {
+    if (data.length < ABIDecoder.OP_CODE_LENGTH) throw new IllegalArgumentException("Data length " + data.length + " must be >= " + ABIDecoder.OP_CODE_LENGTH)
+    util.Arrays.copyOf(data, ABIDecoder.OP_CODE_LENGTH)
+  }
+
+}
+
+trait ABIDecoder[T] {
+  private val decoder = new DefaultFunctionReturnDecoder()
+  val ListOfABIParamTypes: util.List[TypeReference[Type[_]]]
+  lazy val ABIDataParamsLengthInBytes = Type.MAX_BYTE_LENGTH * ListOfABIParamTypes.size()
+
+  def decode(abiEncodedData: Array[Byte]): Try[T] = {
+    Try {
+      val inputParams = ABIDecoder.getArgumentsFromData(abiEncodedData)
```

Comment:
why we should assume that `abiEncodedData` contains both method and data part?
Considering that in the end we wan't to create only the command input/args, it is better to call:
`val inputParams = ABIDecoder.getArgumentsFromData(abiEncodedData)` this outside and to pass the exact class after.
Note: we may use this class not only for fake smart contract inputs. I want it to be designed for a generic purpose.

Reply:
Done